plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java

```diff
@@ -124,22 +124,15 @@ public void onSuccess(StatusResponse result)
                             {
                                 verify(result != null);
 
-                                if (result.getStatusCode() >= 500 && attempt < config.getRetryCount()) {
+                                if (!(result.getStatusCode() >= 200 && result.getStatusCode() < 300) && attempt < config.getRetryCount()) {
```

Comment:
I'm not sure how would retrying any 4xx error ever succeed? Wouldn't we end up retrying until retry attempts are exhausted?

Reply:
Yes, we would retry until attempts are exhausted.

Reply:
I think it has its use-cases (which admittedly are edge-cases) and it's better to be safe than drop data. This doesn't run on the query execution threads so no direct slowdown.